packages/blob/src/index.ts

```diff
@@ -89,6 +80,8 @@ export async function put(
     throw new BlobError('access must be "public"');
   }
 
+  await checkAndRetrieveClientToken(pathname, options);
```

Comment:
Can we write this code to not mutate the options object? While we do not have an eslint rule for this. This is usually not recommended:
1. It makes understanding the code harder (I had to carefully read the method to understand why we're only await ing and not returning any result)
2. In this case, we're also mutating objects the user sends. Consider this use case:

```
// customer code
const defaultPutOptions = { handleBlobUploadUrl };

await put('file1.png', defaultPutOptions); // works
await put('file2.png', defaultPutOptions); // fails, defaultPutOptions now has a token property.
```

Recommendation:

```
const token = isClientPut(options) ? await retrieveClientToken(options) : options.token; 
```

Something similar to this might make the code more obvious and will avoid mutating objects.

Reply:
fair point, I will make it immutable